crates/uv/tests/it/network.rs

```diff
@@ -0,0 +1,154 @@
+use std::env;
+
+use assert_fs::fixture::{FileWriteStr, PathChild};
+use http::StatusCode;
+use serde_json::json;
+use wiremock::matchers::method;
+use wiremock::{Mock, MockServer, ResponseTemplate};
+
+use crate::common::{TestContext, uv_snapshot};
+
+/// Check the simple index error message when the server returns HTTP status 500, a retryable error.
+#[tokio::test]
+async fn simple_http_500() {
+    let context = TestContext::new("3.12");
+
+    let server = MockServer::start().await;
```

Comment:
It looks like you can break this out into a separate function and just return the URI.

Reply:
You've already broken this out in the follow up PR so it may not be necessary here if you've touched all these cases.